internal/core/src/segcore/Utils.cpp

```diff
@@ -869,70 +869,73 @@ ReverseDataFromIndex(const index::IndexBase* index,
 // segcore use default remote chunk manager to load data from minio/s3
 void
 LoadArrowReaderFromRemote(const std::vector<std::string>& remote_files,
-                          std::shared_ptr<ArrowReaderChannel> channel) {
-    try {
-        auto rcm = storage::RemoteChunkManagerSingleton::GetInstance()
-                       .GetRemoteChunkManager();
-        auto& pool = ThreadPools::GetThreadPool(ThreadPoolPriority::HIGH);
-
-        std::vector<std::future<std::shared_ptr<milvus::ArrowDataWrapper>>>
-            futures;
-        futures.reserve(remote_files.size());
-        for (const auto& file : remote_files) {
-            auto future = pool.Submit([rcm, file]() {
+                          std::shared_ptr<ArrowReaderChannel> channel,
+                          milvus::ThreadPoolPriority priority) {
+    auto rcm = storage::RemoteChunkManagerSingleton::GetInstance()
+                   .GetRemoteChunkManager();
+    auto& pool = ThreadPools::GetThreadPool(priority);
+    auto remaining_tasks =
+        std::make_shared<std::atomic<int>>(remote_files.size());
+    auto next_to_push = std::make_shared<std::atomic<size_t>>(0);
+
```

Comment:
we need to keep the order of insert_files for consistency between different fields